### PR TITLE
8254175: Build no-pch configuration in debug mode for submit checks

### PR DESCRIPTION
**Pull request by shipilev**
no-pch configuration is supposed to expose missing include dependencies. But currently it runs with default (release) bits, which misses symbols hidden in debug code. We should consider building it in debug mode.

Attention @rwestberg.

Testing:
  - [x] GH workflow still works, see the builds in the [latest run](https://github.com/shipilev/jdk/actions/runs/293802758)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ✔️ (3/3 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8254175](https://bugs.openjdk.java.net/browse/JDK-8254175): Build no-pch configuration in debug mode for submit checks

### Reviewers
 * [Robin Westberg](https://openjdk.java.net/census#rwestberg) (@rwestberg - Committer)
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/547/head:pull/547`
`$ git checkout pull/547`

### CODE REVIEW

File: .github/workflows/submit.yml [hunk 1]
```diff
@@ -105,13 +105,13 @@ jobs:
         flavor:
           - build release
           - build debug
-          - build hotspot no-pch
+          - build debug hotspot no-pch
```
Comment (rwestberg) [inline .github/workflows/submit.yml:108]:
```suggestion
          - build hotspot no-pch
```

File: .github/workflows/submit.yml [hunk 2]
```diff
@@ -105,13 +105,13 @@ jobs:
...
         include:
           - flavor: build debug
             flags: --enable-debug
             artifact: -debug
-          - flavor: build hotspot no-pch
-            flags: --disable-precompiled-headers
+          - flavor: build debug hotspot no-pch
```
Comment (rwestberg) [inline .github/workflows/submit.yml:113]:
```suggestion
          - flavor: build hotspot no-pch
```

**Review by rwestberg — APPROVED** [review]
Looks good to me! Just a similar comment as in #546, the name is a bit too long now, so perhaps just drop the "debug" from the description?

File: .github/workflows/submit.yml (hunk 1 above)
Comment (shipilev) [reply .github/workflows/submit.yml:108]: Right, we can do that. Let me regenerate both #546 and #547

**Review by rwestberg — APPROVED** [review]
Looks good!

**Comment by shipilev** [comment]
@erikj79, can you take a look as well? Or maybe @dholmes-ora wants to chime in on the sanity of the change?

**Review by erikj79 — APPROVED** [review]

**Comment by shipilev** [comment]
This affects testing for everyone, so I would like someone from hotspot to chime in.

/label hotspot

**Review by dholmes-ora — APPROVED** [review]
Seems reasonable.

**Pull request closed without merging**